pandas/tests/frame/indexing/test_setitem.py

```diff
@@ -1082,7 +1082,7 @@ def test_setitem_partial_column_inplace(self, consolidate, using_array_manager):
         tm.assert_numpy_array_equal(zvals, expected.values)
         assert np.shares_memory(zvals, df["z"]._values)
         if not consolidate:
-            assert df["z"]._values is zvals
+            assert df["z"]._values.base is zvals.base
```

Comment:
this check could pass if both .base attrs are None

Reply:
Yes, good catch (will update that). Now, I also not fully sure why this change was actually needed, will see if I can figure that out.

Reply:
OK, so before we had this piece of code in indexing.py (that was removed in this PR):

```
            if ser._values is orig_values:
                # The setitem happened inplace, so the DataFrame's values
                #  were modified inplace.
                return
```

That ensures that it is actually still an _identical_ array that is stored in `df["z"]`. 

Now this goes through `column_setitem`, where I didn't add such a check, but simply always call `iset` with the values from the SingleBlockManager. When creating the SingleBlockManager with `iget`, this creates a view of the values (`return self.values[i]` in Block.iget), so this is not anymore an identical array, only a viewing array. 

Since the only important guarantee is that it shares the same data (which is already tested on the line above), I am inclined to just remove this additional `assert` for identical arrays.